processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -71,5 +77,63 @@ func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
 	attr.InsertString(conventions.AttributeHostType, meta.InstanceType)
 	attr.InsertString(conventions.AttributeHostName, hostname)
 
+	tags, err := connectAndFetchEc2Tags(meta.Region, meta.InstanceID, d.cfg)
+	if err != nil {
+		return res, fmt.Errorf("failed fetching ec2 instance tags: %w", err)
+	}
+
+	for key, val := range tags {
+		attr.InsertString(tagPrefix + key, val)
+	}
+
 	return res, nil
 }
+
+func connectAndFetchEc2Tags(region string, instanceID string, cfg Config) (map[string]string, error) {
+	if(!cfg.AddAllTags && len(cfg.TagsToAdd) == 0){
+		return nil, nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = sess.Config.Credentials.Get(); err != nil {
+		return nil, err
+	}
+	e := ec2.New(sess)
+
+	return fetchEC2Tags(e, instanceID, cfg)
+}
+
+func fetchEC2Tags(e ec2iface.EC2API, instanceID string, cfg Config) (map[string]string, error) {
+	ec2Tags, err := e.DescribeTags(&ec2.DescribeTagsInput{
+		Filters: []*ec2.Filter{{
+			Name: aws.String("resource-id"),
+			Values: []*string{
+				aws.String(instanceID),
+			},
+		}},
+	})
+	if err != nil {
+		return nil, err
+	}
+	tags := make(map[string]string)
+	for _, tag := range ec2Tags.Tags {
+		if cfg.AddAllTags || contains(cfg.TagsToAdd, *tag.Key) {
```

Comment:
Can we make `cfg.TagsToAdd` a set instead of a list? (Would help with performance.)

Reply:
I dont think a set data structure exists, the alternative would be to use a map. I thought of this and opted for an array because I don't believe performance to be an issue here given that 50 tags is the maximum allowed for an EC2 instance. I also thought this:

```
ec2: 
  tags_to_add:
    - tag1
    - tag2
```

visually makes more sense than this:

```
ec2:
  tags_to_add:
    tag1:
    tag2:
```

Reply:
Yeah, it's fine as a list given our limits. Don't think it really matters how we have it in the config, if we wanted, we could parse the yaml as a list and convert it to a set later. I would be good with leaving it as is and using a list tho.